Emitter/Emitter.cs

```diff
@@ -238,6 +245,30 @@ private void OnMessageReceived(object sender, Messages.MqttMsgPublishEventArgs e
                     return;
                 }
 
+                //Did we receive a response to ExecuteAsync request?
+                if (_requestNames.ContainsKey(e.Topic))
+                {
+                    try
+                    {
+                        var message = Encoding.UTF8.GetString(e.Message);
+                        if (message.Contains("\"req\"")) //done for faster pre-check
+                        {
+                            Hashtable map = (Hashtable)JsonSerializer.DeserializeString(message);
+                            if (map.ContainsKey("req"))
+                            {
+                                if (_waitingRequests.TryRemove(int.Parse(map["req"].ToString()), out var tcs))
+                                {
+                                    if (!(tcs.Task.IsCompleted || tcs.Task.IsCanceled))
```

Comment:
Suggest simplifying all those nested if statements, it affects readability.

Reply:
Sometimes it enhances readability, in my opinion.